src/Passes/Source/Rules/Operands/UnnamedInvoke.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Rules/IOperandPrototype.hpp"
+#include "Rules/Operands/UnnamedInvoke.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    UnnamedInvokePattern::~UnnamedInvokePattern() = default;
```

Comment:
(I'm totally fine with the items below, but some readers may get a sour grimace on their face)

1. Multiple files with 
either the same name ("UnnamedInvoke.cpp") in different dirs ("Operands" and "Notation")
or close names (Phi.hpp and Phi.ipp, Call.hpp and Call.ipp) in the same dirs.
2. It is a bit confusing that the directory is named `Operands` but all the class names defined inside of it end with `Pattern` (e.g. "UnnamedInvoke**Pattern**"). This makes me think that either the directory needs to be renamed to `Patterns` or the classes should be renamed such that the class names end with `Operand` rather than `Pattern` (e.g. "UnnamedInvoke**Operand**").
3. Consider renaming all the files inside of `Operands` folder to match the class names. E.g. the file name is 
"UnnamedInvoke.cpp" but the class name is 
"UnnamedInvoke**Pattern**".
4. Consider renaming .ipp files to .hpp files in "Notation" directory.

Reply:
Yes, I agree that this could need some cleaning. Unfortunately, I will not be able to do that tonight.

Reply:
I've reworked the names.

Regarding 4., I would refrain from renaming it as this practice allows separation of the interface from the implementation for templated classes and/or functions. Some more on this [here](https://stackoverflow.com/questions/19147208/difference-between-using-ipp-extension-and-cpp-extension-files) and [here](https://jmmv.dev/2013/12/header-files-c-ipp-files.html). Instead what I've done is to move the call definition into `Call.hpp`.

Reply:
Thank you or the links. Totally makes sense.